OneCommunityGlobal/HighestGoodNetworkApp
# Vijay Anand - Added search by name functionality feature to Leaderboard Component

# Description
![Screenshot 2024-08-14 at 10 14 45 PM](https://github.com/user-attachments/assets/399bcf29-c3cd-4c05-8754-94e8dae9eec6)


## Related PRS (if any):
To test this frontend PR you need to checkout the development branch in backend.
…

## Main changes explained:
- Added debounced search functionality to search users by name in the leaderboard component table for better user experience.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin/owner user
5. go to the dashboard
6. verify whether you could search user from the leaderboard component table by name.
7. verify this new feature works in [dark mode].
8. verify the responsiveness of this new feature.
9. verify other functionalities of the leaderboard components are unaffected by the new change - like viewing the user dashboard by clicking on the status and viewing the user profile by clicking the user name.

## Screenshots or videos of changes:
**Testing the new feature**
https://www.loom.com/share/2264d26b08994dfd9e0b8ef5f2414c15?sid=b416506a-873e-49d4-bfba-e49bb6217c80

## Note:
- Start testing the functionality only after the entire dashboard is loaded.
- Search input will be cleared every time the admin/owner user performs view user dashboard or view user profile because the component gets unmounted.
- Any other existing dashboard UI issues are not related to the leaderboard component.
